### PR TITLE
feat(shorebird_cli): support using linker from local engine

### PR DESCRIPTION
**Pull request by felangel**
## Description

- feat(shorebird_cli): support using linker from local engine

## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/executables/aot_tools.dart [hunk 1]
```diff
@@ -11,18 +12,23 @@ AotTools get aotTools => read(aotToolsRef);
...
-    final executable = p.join(
-      cache.getArtifactDirectory(executableName).path,
-      executableName,
+    final executable = shorebirdArtifacts.getArtifactPath(
+      artifact: ShorebirdArtifact.aotTools,
     );
 
+    if (p.extension(executable) == '.dart') {
+      return process.run(
+        'dart',
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/executables/aot_tools.dart]: This probably deserves a TODO to use the Dart installed.  You could also use Platform.resolvedExecutable or Platform.exectuable: https://api.dart.dev/stable/3.2.3/dart-io/Platform/resolvedExecutable.html

File: packages/shorebird_cli/lib/src/shorebird_artifacts.dart [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    switch (artifact) {
+      case ShorebirdArtifact.aotTools:
+        return _aotToolsFile.path;
+    }
+  }
+
+  File get _aotToolsFile {
+    return File(
+      p.join(
+        engineConfig.localEngineSrcPath!,
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/shorebird_artifacts.dart:134]: We might end up with this as engineConfig.path(....) to avoid and save some chars. 🤷

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/lib/src/cache.dart [hunk 3]
```diff
@@ -47,6 +49,18 @@ final cacheRef = create(Cache.new);
...
+/// A class that manages the artifacts cached by Shorebird.
+/// This class handles fetching and unpacking artifacts from various sources.
+///
+/// To access specific artifacts, it's generally recommended to use
+/// [FlutterCachedArtifacts] or [ShorebirdCachedArtifacts] respectively
+/// since they take into account the current Shorebird environment.
+///
+/// See also:
+/// * [FlutterCachedArtifacts], provides a list of Flutter cached artifacts.
+/// * [ShorebirdCachedArtifacts], provides a list of Shorebird cached artifacts.
```
Comment (bryanoltman) [inline packages/shorebird_cli/lib/src/cache.dart]: Here (or somewhere in this PR), it'd be good to spell out explicitly what we mean when distinguishing between a Flutter artifact and a shorebird artifact (specifically, flutter artifacts exist in stock flutter but may have been modified by our fork, whereas shorebird artifacts are only exist in the shorebird world and have no mainline flutter version).

**Review by bryanoltman — APPROVED** [review]

File: packages/shorebird_cli/lib/src/executables/aot_tools.dart (hunk 1 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/executables/aot_tools.dart]: Fixed

File: packages/shorebird_cli/lib/src/cache.dart (hunk 3 above)
Comment (felangel) [reply packages/shorebird_cli/lib/src/cache.dart]: I consolidated the two into `ShorebirdArtifacts` 👍

**Pull request merged**